debezium-core/src/main/java/io/debezium/data/SpecialValueDecimal.java

```diff
@@ -119,4 +128,48 @@ else if (!decimalValue.equals(other.decimalValue))
             return false;
         return true;
     }
+
+    /**
+     * Returns a {@link SchemaBuilder} for a decimal numbers depending on {@link JdbcValueConverters.DecimalMode}.
+     * You can use the resulting SchemaBuilder
+     * to set additional schema settings such as required/optional, default value, and documentation.
+     *
+     * @param mode - the mode in which the number should be encoded
+     * @param scale - scale of the decimal
+     * @return the schema builder
+     */
+    public static SchemaBuilder builder(DecimalMode mode, int scale) {
+        switch (mode) {
+        case DOUBLE:
+            return SchemaBuilder.float64();
+        case PRECISE:
+            return Decimal.builder(scale);
+        case STRING:
+            return SchemaBuilder.string();
+        }
+        throw new IllegalArgumentException("Unknown decimalMode");
+    }
+
+    public static Object fromLogical(SpecialValueDecimal value, DecimalMode mode, String columnName) {
+        if (value.getDecimalValue().isPresent()) {
+            switch (mode) {
+            case DOUBLE:
+                return value.getDecimalValue().get().doubleValue();
+            case PRECISE:
+                return value.getDecimalValue().get();
+            case STRING:
+                return value.getDecimalValue().get().toString();
+            }
+            throw new IllegalArgumentException("Unknown decimalMode");
+        }
+
+        // special values (NaN, Infinity) can only be expressed when using "string" encoding
+        if (mode == DecimalMode.STRING) {
```

Comment:
@jpechane, I'm a bit confused, the special values are expressable with `Double`, too. Is this missed here or is it in fact just handled elsewhere?

Reply:
@gunnarmorling It is really suspicious - I'll first check why the tests has not caught the issue.

Reply:
@gunnarmorling Thanks for soptting, there were missing asserts in test and that's why this bug has passed. Bot issue and tests have been fixed.